### PR TITLE
impl(storage): always set content-length and fix LockRetentionPolicy parameters

### PR DESCRIPTION
**Pull request by scotthart**
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/9681)

### CODE REVIEW

File: google/cloud/internal/curl_rest_client.cc [hunk 1]
```diff
@@ -52,7 +52,9 @@ Status MakeRequestWithPayload(
     }
     encoded_payload = impl.MakeEscapedString(concatenated_payload);
     if (!encoded_payload.empty()) {
```
Comment (dbolduc) [inline google/cloud/internal/curl_rest_client.cc:54]: can we get rid of this if altogether? does `encoded_payload.empty() => encoded_payload.size() == 0`?

**Review by scotthart — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/9681)* status: 0 of 2 files reviewed, 1 unresolved discussion (waiting on @dbolduc)

___
*[`google/cloud/internal/curl_rest_client.cc` line 54 at r1](https://reviewable.io/reviews/googleapis/google-cloud-cpp/9681#-N9EJbUrRAMDY63Izis6-r1-54:-N9EbYZ23JwWsOTvO_n9:bb74ce2) ([raw file](https://github.com/googleapis/google-cloud-cpp/blob/a830e01c17b253284ac450ebf5c53e431f83e88f/google/cloud/internal/curl_rest_client.cc#L54)):*
<details><summary><i>Previously, dbolduc (Darren Bolduc) wrote…</i></summary><blockquote>

can we get rid of this if altogether? does `encoded_payload.empty() => encoded_payload.size() == 0`? 
</blockquote></details>

Sure

File: google/cloud/internal/curl_rest_client.cc [hunk 2]
```diff
@@ -62,9 +60,8 @@ Status MakeRequestWithPayload(
   for (auto const& p : payload) {
     content_length += p.size();
   }
-  if (content_length > 0) {
-    impl.SetHeader(absl::StrCat("content-length: ", content_length));
-  }
+
+  impl.SetHeader(absl::StrCat("Content-Length: ", content_length));
```
Comment (coryan) [inline google/cloud/internal/curl_rest_client.cc:64]: Please make sure we test both with zero and non-zero values for `content_length`.

**Review by coryan — APPROVED** [review]

**Review by scotthart — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/9681)* status: 0 of 2 files reviewed, 2 unresolved discussions (waiting on @coryan and @dbolduc)

___
*[`google/cloud/internal/curl_rest_client.cc` line 64 at r2](https://reviewable.io/reviews/googleapis/google-cloud-cpp/9681#-N9EJbUrRAMDY63Izis6-r2-64:-N9HvhaN4KWM2TGzBlFd:b-t3d93y) ([raw file](https://github.com/googleapis/google-cloud-cpp/blob/37ba54cf3286d30b865ce6a67a3447016ec2f81c/google/cloud/internal/curl_rest_client.cc#L64)):*
<details><summary><i>Previously, coryan (Carlos O'Ryan) wrote…</i></summary><blockquote>

Please make sure we test both with zero and non-zero values for `content_length`.
 
</blockquote></details>

Failures in the integration-production build where the content-length needed to be set to zero spurred this change. Test cases where content-length was non-zero were already passing.

**Pull request merged**